torch/distributed/fsdp/_optim_utils.py

```diff
@@ -219,7 +219,7 @@ def _communicate_optim_state(
         # value directly
         else:
             if _is_zero_dim_tensor(value):
-                zero_dim_tensor_state[state_name] = value
+                zero_dim_tensor_state[state_name] = value.detach().clone()
             else:
                 non_tensor_state[state_name] = value
```

Comment:
how about this one?

Reply:
This is a non-tensor case. I don't think we have to worry about Python object (e.g., int, float)?